vllm/v1/worker/tpu_model_runner.py

```diff
@@ -587,17 +620,17 @@ def _execute_mm_encoder(self, scheduler_output: "SchedulerOutput"):
                 encoder_outputs.append(output)
 
         # Cache the encoder outputs.
-        for (req_id, input_id, pos_info), output in zip(
+        # NOTE (NickLucche) here we diverge from logic in other runners, as we
+        # assume to only have whole mm items to process. Hence we avoid the
+        # intrinsic dynamism that `scatter_mm_placeholders` introduces.
+        for (req_id, input_id, _), output in zip(
                 req_ids_pos,
                 encoder_outputs,
         ):
             if req_id not in self.encoder_cache:
                 self.encoder_cache[req_id] = {}
 
-            self.encoder_cache[req_id][input_id] = scatter_mm_placeholders(
-                output,
-                is_embed=pos_info.is_embed,
-            )
+            self.encoder_cache[req_id][input_id] = output
```

Comment:
Should we do an assert on pos_info.is_embed to make sure it is all true? We need to check something that we aren't running a model that requires this

Reply:
this makes sense

Reply:
yeah it does

Reply:
And this is an issue because we aren't in a graph at this point? Maybe we should make this a small graph or wrap in torch.compile. I feel bad not having a check for this

Reply:
Maybe you could check in the input processor on the CPU where is_embed is created

Reply:
For the model we're interesteed in supporting atm, it appears we can just assert None here https://vllm-dev.slack.com/archives/C07QCGVDNUF/p1744361689005189?thread_ts=1744359817.739719&cid=C07QCGVDNUF. I can update the description with a benchmark run.